subsys/bluetooth/host/gatt.c

```diff
@@ -3475,10 +3475,17 @@ bool bt_gatt_is_subscribed(struct bt_conn *conn,
 
 	/* Check if attribute is a characteristic declaration */
 	if (!bt_uuid_cmp(attr->uuid, BT_UUID_GATT_CHRC)) {
-		struct bt_gatt_chrc *chrc = attr->user_data;
+		uint8_t properties;
+		ssize_t len;
+
+		/* The charactestic properties is the first byte of the attribute value */
+		len = attr->read(NULL, attr, &properties, 1, 0);
```

Comment:
Aren't you making a new assumption here? 

If we cannot assume that the user data is a `struct bt_gatt_chrc`, can we then assume that the first octet is the properties? 

If `BT_GATT_CHARACTERISTIC` was used to declare the characteristic, the first assumption holds. If it was not, then this assumption doesn't hold either, does it?

Reply:
I disagree, and would at the very least say that the new assumption is more precise. The user data field is described as varying: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/bluetooth/gatt.h#L267, and is not part of the GATT specification. The `read` method is a must-have for readable attributes, and returns the Attribute Value. According to Table 3.4 of the GATT Profile this has the specific structure where the properties is the first octet (since we're dealing specifically with the characteristic declaration).

Reply:
(I should perhaps add a check to see whether `read` is NULL or not: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/bluetooth/gatt.h#L256)

Reply:
> I should perhaps add a check to see whether `read` is NULL or not

Yes, that's a good idea. This is potentially application provided and could (incorrectly) be NULL, and we don't want to crash. Note, it's incorrect for this to be NULL because a CCCD must be readable.